File: src/main/kotlin/ch/wngr/bookstore/services/BasketService.kt
package ch.wngr.bookstore.services

import ch.wngr.bookstore.models.BasketDto
import ch.wngr.bookstore.models.SaleDTO
import org.springframework.http.ResponseEntity

interface BasketService {
    fun addBooksToBasket(saleDTOs: List<SaleDTO>)
    fun getBaskets(open: Boolean): ResponseEntity<List<BasketDto>>
    fun closeBasket(basketID: Int): ResponseEntity<BasketDto>
}

File: src/frontend/src/service/BasketService.ts
import handleErrors from '@/helpers/httpHelper';
import {BasketDto} from "@/models/BasketDto";

class BasketService {

    getBaskets(open: boolean) : Promise<BasketDto[]> {
        return fetch(`/api/basket?${new URLSearchParams({open: open.toString()})}`)
            .then(handleErrors)
            .then(res => res.json())
    }

    closeBasket(basketId: string) : Promise<BasketDto> {
        const requestOptions = {
            method: "DELETE",
        }
        return fetch("/api/basket/" + basketId, requestOptions)
            .then(handleErrors)
            .then(res => res.json())
    }
}

export default new BasketService()